### PR TITLE
Improve formatting of configuration.md

### PR DESCRIPTION
**Pull request by mstanleyjones**
Started out to fix a formatting bug seen at https://docs.docker.com/registry/configuration/#/log, but ended up reformatting the whole file and doing some light rewriting. I do not believe I changed any "facts".

@stevvooe please nominate other reviewers.

### CODE REVIEW

**Comment by sanscontext** [comment]
The diff looks good, but I haven't been able to untangle my conflicting docker installs to find a way to build and preview it. In the mean time, attaching these issues, which will be blessedly fixed when this is done. :)

Closes docker/docker.github.io/issues/986
Closes docker/docker.github.io/issues/983
Closes docker/docker.github.io/issues/968
Closes docker/docker.github.io/issues/927
Closes docker/docker.github.io/issues/899

File: docs/configuration.md [hunk 1]
```diff
@@ -1966,67 +1078,54 @@ http:
...
+S3 storage driver.
+
+This table describes the configuration:
+
+| Parameter | Description                                           |
+|-----------|-------------------------------------------------------|
+| `name`    | The name of the storage middleware. `cloudfront` is an accepted value. |
+| `disabled`| Set to `true` to easily disable the middleware.       |
+| `options` | A set of key-value options to configure the middleware. The rest of the rows of this table are the options. |
+| `baseurl` | The CloudFront base URL.                              |
```
Comment (dmcgowan) [inline docs/configuration.md]: This section is still kind of confusing.  Since these are `cloudfront` specific options they should probably be broken out into a completely separate section for `cloudfront` options. We do have another PR open for another type of middleware so it is probably best to break this up.

**Comment by dmcgowan** [comment]
LGTM

Commented on middleware section but that could be broken out later as well.

**Comment by dmcgowan** [comment]
BTW thanks @mstanleyjones for making this change, its a pretty big cleanup and a huge improvement over what was there before.

File: docs/configuration.md (hunk 1 above)
Comment (mstanleyjones) [inline docs/configuration.md]: Actually those `cloudfront` options are already broken out elsewhere. I reworked this section a little to remove the table (but keep the YAML) and to point to the `cloudfront` section of the file for more details. Let me know what you think.

**Comment by ArcticSnowman** [comment]
The page https://docs.docker.com/registry/configuration/ is badly broken,  this fix needs to be pushed out.
